beets/importer.py

```diff
@@ -1490,7 +1500,8 @@ def albums_in_dir(path):
         if collapse_paths:
             if (not collapse_pat and collapse_paths[0] in ancestry(root)) or \
                     (collapse_pat and
-                     collapse_pat.match(os.path.basename(root))):
+                     collapse_pat.match(os.path.basename(root))) or \
+                    is_subdir_of_any_in_list(root, collapse_paths):
```

Comment:
Good point, I'm not completely sure about removing "not collapse_pat and", but indeed, if that's "removable", then it can be simplified. In my tests that seems to work fine at least.